src/components/notifications/NotificationsMenu.js

```diff
@@ -149,6 +133,57 @@ function NotificationsMenu(props) {
         },
     });
 
+    const { isMarked } = useQuery({
+        queryKey: NOTIFICATIONS_MARK_ALL_SEEN_KEY,
+        queryFn: markAllSeen,
+        config: {
+            onSuccess: () => {
+                setAllNotificationsSeen();
+            },
+            onError: (error) => {
+                console.log("Error marking all notifications as Seen");
+                // *** Will add this after understanding HOC better. ***
+                // showErrorAnnouncer(
+                //     t("errorMarkAsSeen", {
+                //         count: notifications.length,
+                //     }),
+                //     error
+                // );
+            },
+        },
+    });
+
+    const setAllNotificationsSeen = () => {
+        console.log(isMarked);
+        setNotifications(
+            notifications?.map((item) => {
+                const tempItem = item;
+                tempItem.seen = true;
+                return tempItem;
+            })
+        );
+    };
+
+    const handleClose = () => {
+        setAnchorEl();
+    };
+
+    const handleMarkAllAsSeenClick = () => {
+        // set badge value as 0.
+        setUnSeenCount(0);
```

Comment:
We should reset this count only after calling the service and getting a success response. Right now I get `404` for `/api/notifications/mark-all-seen` cos the corresponding handler in `/src/server/api/notifications.js` is missing.

Reply:
I think I may have helped cause some confusion here 😖 .  The other day he asked me if an API call failing was expected and in the screenshot was the error page for `/api/bootstrap`.  I said that the bootstrap endpoint is expected to fail because iRods was currently down because I thought that's what he was asking me and I think he took my response as his API call is expected to fail. Sorry Flynn!